src/Runner.Worker/action_yaml.json

```diff
@@ -120,7 +120,22 @@
                 "hashFiles(1,255)"
             ],
             "sequence": {
-                "item-type": "any"
+                "item-type": "composite-step"
+            }
+        },
+        "composite-step": {
+            "mapping": {
+                "properties": {
+                    "name": "string-steps-context",
+                    "id": "non-empty-string",
+                    "run": {
+                        "type": "string-steps-context",
+                        "required": true
+                    },
+                    "env": "step-env",
+                    "working-directory": "string-steps-context",
+                    "shell": "string-steps-context"
```

Comment:
`shell` in `workflow-v1.0.json` is `non-empty-string` (expressions not allowed). Let's match that value. Currently you have `string-steps-context` instead. Good to test after making the change and verify expression not allowed for `shell` value.

Reply:
shell is evaluated unfortunately due to the if condition in ScriptHandler: https://github.com/actions/runner/blob/main/src/Runner.Worker/Handlers/ScriptHandler.cs#L255-L271. Demo: https://github.com/ethanchewy/testing-actions/runs/915535061?check_suite_focus=true. 

Will put in fix soon.